site/README.md

```diff
@@ -0,0 +1,54 @@
+# aip.dev static site generator
+
+Welcome to our NIH static site generator.
+
+**Note:** If you do not care about this and just want to write or modify AIPs,
+see `CONTRIBUTING.md`.
+
+## Why?
+
+GitHub Pages normally automatically builds documentation with [Jekyll][], but
+as the AIP system has grown, we are beginning to reach the limits of what
+Jekyll can handle:
+
+- AIP adoption is handled through fork-and-merge and top-down configuration
+  files will lead to repetitive merge conflicts.
+- Our grouping and listing logic has grown complicated, and had to be
+  maintained using complex and error-prone Liquid templates.
+- Jekyll is extensible but GitHub requires specific Jekyll plugins, meaning we
+  can not use off-the-shelf solutions for planned features (e.g. tabbed code
+  examples).
+- Lack of meaningful build CI caused failures.
+- Working with the development environment was (really) slow.
+
+## Grumble. Fine. How does it work?
+
+The app is essentially split up into models (`generator/models.py`) and
+templates (`templates/`). The models file provides data classes that are sent
+to the templates and which have convenient accessors for template needs (title,
+relative URI, etc.)
+
+There are three models that matter:
+
+- **Site:** A singleton (ish) that provides access to all scopes, AIPs, and
```

Comment:
That "(ish)" has me curious. Do you have more info on that?

Reply:
It is kind of a sloppy singleton right now. The `Site.load()` method gives you have an object with (lazily loaded) references to all AIPs, categories, and other pages. But there is no mechanic to force you to always get back the exact same object that was loaded the first time.

This is likely something worth fixing.